feat: add support for Device Hardware json and svg graphics

jamesarich opened the pull request:
Adds device_hardware.json, DeviceHardware model, svg device graphics, and convenience methods to parse the json and map it to DeviceHardware model and image resource.

Allows for better hardware device display names, graphics, and indication of support.

![image](https://github.com/user-attachments/assets/bdd5aa42-c766-4ff1-84c1-70fc42a06631)

andrekir commented:
Please move the images from the detail section to a header at the top of the screen. The header should place the image in focus without overwhelming the rest of the screen, so it also needs to work well with both horizontal and vertical images. Maybe use a complementary background with the hardware name under the image.

jamesarich commented:
@andrekir - i moved the relevant device info up to a new `Device` section above the `Details`. I also restricted the device image size and gave it a circular background with the node color (alpha 50%).

andrekir commented on app/src/main/java/com/geeksville/mesh/ui/NodeDetail.kt:
```diff
@@ -196,7 +214,51 @@ private fun NodeDetailRow(label: String, icon: ImageVector, value: String) {
+) {
+    val context = LocalContext.current
+    val deviceHardware =
+        DeviceHardware.getDeviceHardwareFromHardwareModel(context, node.user.hwModel)
```
this can move to the ViewModel and use app context. also needs some exception handling.

jamesarich commented:
@andrekir - I moved the DeviceHardware and image res functions to MetricsViewModel, and added some State variables.

The DeviceHardware json parsing now has exception handling and is cached in the model to avoid expensive rereads.

I also updated the image res function to use `when` mapping from HardwareModel enum to Drawable resource id. This function will need to be updated when new images are added.
